tests/Properties.hs

```diff
@@ -82,9 +82,29 @@ prop_unsafeTail xs = not (P.null xs) ==> P.tail xs === P.unsafeTail xs
 prop_unsafeLast xs = not (P.null xs) ==> P.last xs === P.unsafeLast xs
 prop_unsafeInit xs = not (P.null xs) ==> P.init xs === P.unsafeInit xs
 
+prop_lines_empty_invariant =
+     True === case LC.lines (LC.pack "\nfoo\n") of
+        Empty : _ -> True
+        _         -> False
```

Comment:
Could we also have a randomized test that uses `invariant` or `checkInvariant` to check the output of `lines`? I was expecting this test to exist already, but I couldn't find it.

Reply:
Could that be a separate PR, or would you care to propose the test?  It'd take me some scarce cycles to *page in* (or acquire) the knowhow of how to construct randomised tests for this, I don't use the facilities in question sufficiently often in real life...

Reply:
Yeah, this can be done separately. To generate the test input, I'd simply use some `arbitrary` `LazyByteString`'s, interleaved with additional newlines.

Reply:
I have opened #564 to track this.